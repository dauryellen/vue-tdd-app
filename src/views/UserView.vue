<template>
  <div>
    <v-user-search-form />
    <v-user-profile />
  </div>
</template>

<script>
import VUserSearchForm from "../components/VUserSearchForm";
import VUserProfile from "../components/VUserProfile";

export default {
  name: "UserView",
  components: {
    VUserSearchForm,
    VUserProfile,
  },
};
</script>

<style></style>
