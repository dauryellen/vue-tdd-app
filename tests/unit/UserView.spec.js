import { shallowMount } from "@vue/test-utils";
import UserView from "../../src/views/UserView.vue";
import VUserSearchForm from "../../src/components/VUserSearchForm";
import VUserProfile from "../../src/components/VUserProfile";

describe("UserView", () => {
  it("renders the component", () => {
    //arrange
    const wrapper = shallowMount(UserView);

    //assert
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders main child components", () => {
    //arrange
    const wrapper = shallowMount(UserView);
    const userSearchForm = wrapper.findComponent(VUserSearchForm);
    const userProfile = wrapper.findComponent(VUserProfile);

    //assert
    expect(userSearchForm.exists()).toBe(true);
    expect(userProfile.exists()).toBe(true);
  });
});
