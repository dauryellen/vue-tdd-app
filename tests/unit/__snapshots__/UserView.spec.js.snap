// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserView renders the component 1`] = `
<div>
  <v-user-search-form-stub></v-user-search-form-stub>
  <v-user-profile-stub></v-user-profile-stub>
</div>
`;
